Dockerfile/Dockerfile

```diff
@@ -0,0 +1,71 @@
+# Ubuntu Trusty Docker file for base TileDB Dependencies
+
+# This Dockerfile copies files from the parent directory, requiring a build
+# context from the parent directory. From `./..`, execute the following to
+# build:
+# docker build -f Dockerfile/Dockerfile -t tiledb:dev .
+#
+# Use the build arg 'enable' to configure optional TileDB components, e.g.:
+#   docker build -f Dockerfile/Dockerfile --build-arg enable=s3,hdfs -t tiledb:dev
+
+# Ubuntu Trusty
+FROM ubuntu:18.04
+
+# Optional components to enable (defaults to empty).
+ARG enable
+
+# Setup home environment
+RUN useradd tiledb
+RUN mkdir /home/tiledb && chown tiledb /home/tiledb
+ENV HOME /home/tiledb
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    wget \
+    unzip \
+    git \
+    cmake \
+    python3 \
+    python3-dev \
+    python3-pip \
+    libssl-dev \
+    && apt-get clean \
+    && apt-get purge -y \
+    && rm -rf /var/lib/apt/lists*
+
+RUN cd /tmp \
+    && wget https://github.com/Kitware/CMake/releases/download/v3.17.2/cmake-3.17.2-Linux-x86_64.tar.gz \
+    && tar -xzf cmake-3.17.2-Linux-x86_64.tar.gz \
+    && cp -R cmake-3.17.2-Linux-x86_64/bin /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/doc /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/man /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/share /usr/
+
+# Copy TileDB source files from the build context to `/home/tiledb/TileDB``
+COPY CMakeLists.txt /home/tiledb/TileDB/CMakeLists.txt
+COPY bootstrap /home/tiledb/TileDB/bootstrap
+COPY cmake /home/tiledb/TileDB/cmake
+COPY examples /home/tiledb/TileDB/examples
+COPY external /home/tiledb/TileDB/external
+COPY scripts /home/tiledb/TileDB/scripts
+COPY test /home/tiledb/TileDB/test
+COPY tiledb /home/tiledb/TileDB/tiledb
+
+# Install TileDB
+RUN cd /home/tiledb/TileDB \
+    && mkdir build \
+    && cd build \
+    && ../bootstrap --enable-debug --enable-verbose --enable-azure --enable-s3 --enable-serialization --enable=${enable} \
+    && make -j2 \
```

Comment:
Should this be `$(nproc)` to allow for more parallelism?

Reply:
I tried this a few times locally and it would fail with an odd error:
```
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[5]: *** [tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/build.make:934: tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/sm/query/reader.cc.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[4]: *** [CMakeFiles/Makefile2:315: tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/all] Error 2
make[3]: *** [Makefile:141: all] Error 2
make[2]: *** [CMakeFiles/tiledb.dir/build.make:122: tiledb-prefix/src/tiledb-stamp/tiledb-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:352: CMakeFiles/tiledb.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

Leaving as is, probably a resource problem